src/clj/ataru/hakija/hakija_routes.clj

```diff
@@ -394,24 +372,29 @@
         (hakija-application-service/create-new-secret-and-send-link
           koodisto-cache tarjonta-service organization-service ohjausparametrit-service
           job-runner
-         (:old-secret request))
+          (:old-secret request))
         (response/ok {})))
-    (api/GET "/tutkinnot/:henkilo-oid" []
-      :path-params [henkilo-oid :- s/Str]
+    (api/GET "/omat-tutkinnot" [:as request]
+      :summary "Returns exams from Koski for strongly authenticated applicant"
       :return [koski-schema/AtaruKoskiTutkintoResponse]
-      (if-let [suoritukset (koski/get-tutkinnot-for-oppija koski-service henkilo-oid)]
-        (response/ok
-          (map (fn [suoritus] (-> {}
-                                  (assoc :tutkintonimi (get-in suoritus [:koulutusmoduuli :tunniste :nimi]))
-                                  (assoc :koulutusohjelmanimi (get-in suoritus [:tyyppi :nimi]))
-                                  (assoc :toimipistenimi (get-in suoritus [:toimipiste :nimi]))
-                                  (assoc :valmistumispvm (get-in suoritus [:vahvistus :päivä]))
-                                  (assoc :koulutustyyppi
-                                         (select-keys
-                                           (get-in suoritus [:koulutusmoduuli :koulutustyyppi])
-                                           [:koodistoUri :koodiarvo]))))
-               (flatten (map :suoritukset (:opiskeluoikeudet suoritukset)))))
-        (response/not-found {})))
+      (let [oppija-session (get-in request [:cookies "oppija-session" :value])
+            session (oss/read-session oppija-session)]
+         (if-let [henkilo-oid (get-in session [:data :person-oid])]
+          (if-let [suoritukset (koski/get-tutkinnot-for-oppija koski-service henkilo-oid)]
+            (response/ok
+              (map (fn [suoritus] (-> {}
+                                      (assoc :tutkintonimi (get-in suoritus [:koulutusmoduuli :tunniste :nimi]))
+                                      (assoc :koulutusohjelmanimi (get-in suoritus [:tyyppi :nimi]))
+                                      (assoc :toimipistenimi (get-in suoritus [:toimipiste :nimi]))
+                                      (assoc :valmistumispvm (get-in suoritus [:vahvistus :päivä]))
+                                      (assoc :koulutustyyppi
+                                             (select-keys
+                                               (get-in suoritus [:koulutusmoduuli :koulutustyyppi])
+                                               [:koodistoUri :koodiarvo]))))
```

Comment:
Tää voisi ehkä olla siistimpi ilman nuolifunktiota ja associa, eli ihan vaan suoraan 
{:tutkintonimi (get-in...)
 :koulutusohjelmanimi (get-in...)
 ...jne}